kernel/kernel-impl/src/main/java/org/sakaiproject/util/impl/ValueEncryptionServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+package org.sakaiproject.util.impl;
+
+import java.security.AlgorithmParameters;
+import java.security.SecureRandom;
+import java.security.spec.KeySpec;
+import java.util.Base64;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.sakaiproject.util.api.ValueEncryptionService;
+
+/**
+ * This class provides encryption/decryption services. Service is thread safe.
+ */
+public class ValueEncryptionServiceImpl implements ValueEncryptionService {
+
+    private static Log M_log = LogFactory.getLog(ValueEncryptionServiceImpl.class);
+
+    // The encryption key.
+    private String key;
+
+    public void setKey(String key) {
+        this.key = key;
+    }
+    
+    public void init(){}
+
+    /* (non-Javadoc)
+     * @see org.sakaiproject.util.impl.ValueEncryptionService#encrypt(java.lang.String)
+     */
+    public String encrypt(String value) {
+    	try {
+	    	//generate a new random SALT
+			SecureRandom random = new SecureRandom();
```

Comment:
You don't want to continually be getting a new `SecureRandom` as it can block if a server runs out of entropy. It should be a member variable. `Random` implementation are thread safe.